sessionize/app/src/main/java/co/touchlab/sessionize/NotificationsApiImpl.kt

```diff
@@ -0,0 +1,108 @@
+package co.touchlab.sessionize
+
+import android.app.AlarmManager
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import app.sessionize.touchlab.lib.R
+import co.touchlab.sessionize.api.NotificationsApi
+import co.touchlab.sessionize.platform.AndroidAppContext
+
+
+class NotificationsApiImpl : NotificationsApi {
+
+    private val notificationPublisher: BroadcastReceiver = NotificationPublisher()
+
+    override fun createLocalNotification(title:String, message:String, timeInMS:Long, notificationId: Int) {
+        // Building Notification
+        val channelId = AndroidAppContext.app.getString(R.string.notification_channel_id)
+        val builder = NotificationCompat.Builder(AndroidAppContext.app, channelId)
+                .setSmallIcon(R.drawable.baseline_insert_invitation_24)
+                .setContentTitle(title)
+                .setContentText(message)
+                .setWhen(timeInMS)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setCategory(NotificationCompat.CATEGORY_REMINDER)
+
+
+        // Building Intent wrapper
+        val intent = Intent().also { intent ->
```

Comment:
Optional: if you use `.apply` rather than `.also` then you don't need an intent parameter